src/test/path-transformers_test.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/// <reference path="../../node_modules/@types/mocha/index.d.ts" />
+
+import {assert} from 'chai';
+import {join as pathJoin, sep as pathSeparator} from 'path';
+
+import {isPlatformWindows, pathFromUrl, urlFromPath} from '../path-transformers';
+
+const WindowsRootPath = 'C:\\Users\\TEST_USER\\TEST_ROOT';
+const MacRootPath = '/Users/TEST_USER/TEST_ROOT';
+const RootPath = isPlatformWindows() ? WindowsRootPath : MacRootPath;
+
+suite('pathFromUrl()', () => {
+
+  test('creates a filesystem path using the platform separators', () => {
+    const otherSeparator = pathSeparator === '/' ? '\\' : '/';
+    const path = pathFromUrl(RootPath, '/some/url/pathname');
+    assert.include(path, pathSeparator);
+    assert.notInclude(path, otherSeparator);
+  });
+
+  test('returns a path if url is absolute', () => {
+    const path = pathFromUrl(RootPath, '/absolute/path');
+    assert.equal(path, pathJoin(RootPath, 'absolute', 'path'));
```

Comment:
Chatted it out in person. Doing bidirectional mapping between document urls and loader urls is still a fuzzy idea. I think it's something that we'll need to do, but we need to do some more exploration and requirements gathering first.